examples/flex_plot/common/flex_common.d

```diff
@@ -0,0 +1,300 @@
+module flex_common;
+
+import mir.random.flex : FlexInterval;
+
+shared static this() {
+    import pyd.pyd : py_init;
+    py_init();
+}
+
+/**
+CFlex glues the Flex algorithm to plotting code and allows to visualize
+the resulting distribution and its hat and squeeze plots.
+This is only intended for testing and visualizing the Flex algorithm.
+*/
+struct CFlex(S)
+{
+    int n;
+    string plotDir;
+    S rho = 1.1;
+    bool plotHistogram;
+    S stepSize = 0.005;
+
+    /**
+    Params:
+        n = number of samples
+        plotDir = root directory for the all plots
+        rho = efficiency of the Flex algorithm
+        plotHistogram = whether a histogram should be plotted too. If false no
+                        values will be sampled and only the hat and squeeze function
+                        will be plotted
+        stepSize = step size of the points in the plot
+    */
+    this(int n, string plotDir = "plots", S rho = 1.1, bool plotHistogram = true, S stepSize = 0.005)
+    {
+        import std.file : exists, mkdir;
+        if (!plotDir.exists)
+            plotDir.mkdir;
+
+        this.n = n;
+        this.plotDir = plotDir;
+        this.rho = rho;
+        this.plotHistogram = plotHistogram;
+        this.stepSize = stepSize;
+    }
+
+    // @@@BUG@@@ template injection doesn't work with opCall
+    /**
+    Creates a Flex instance given the input parameter and plots it.
+    Refer to the Flex documentation for the parameters.
+
+    Params:
+        name = name of the plot
+        f0 = log-density distribution
+        f1 = first derivative of f0
+        f1 = first derivative of f1
+        c = T_c family to use for the transformation
+        cs = T_c families to use for the transformation
+        points = non-overlapping partitioning with at most one inflection point per interval
+        left = left plotting border
+        right = right plotting border
+    */
+    auto plot(string name, in S function(S) f0, in S function(S) f1, in S function(S) f2,
+         S c, S[] points, S left = -3, S right = 3) const
+    {
+        auto cs = new S[points.length - 1];
+        foreach (ref d; cs)
+            d = c;
+        return plot(name, f0, f1, f2, cs, points, left, right);
+    }
+
+    /// ditto
+    auto plot(string name, in S function(S) f0, in S function(S) f1, in S function(S) f2,
```

Comment:
I would prefer if I can pass in a `Tinflex` instance and don't  need to create these overloads.
This however requires that I am allowed to expose `pdf` for the `Flex` struct ;-)